### ISSUE TITLE
Schedule Script Issue

### ISSUE THREAD

**Issue by emoeckel** [ironmansoftware/powershell-universal#3736]
### Version

5.0.6

### Severity

High

### Environment

msi

### Steps to Reproduce

I have an issue with a script that writes data to Cache, it does it without any error when I run it manually. But if I schedule it and trigger it I get an error "Specify a computer name or use the Connect-PSUServer command."

### Expected behavior

```console
Just works
```

### Actual behavior

```console
Specify a computer name or use the Connect-PSUServer command.
```

**Comment by emoeckel** [comment]
So to add to this, I think this is a weird issue with the cache. I ran the script manually without triggering the schedule. It finishes without errors, but when I try to retrieve the data with the API call that I have I get the same error.

Script section where it write to cache.

`Set-PSUCache -Key "Cache_RVTools_vInfo" -Value ($RvTools_Import_vInfo) -Persist -AbsoluteExpirationFromNow ([TimeSpan]::FromHours(25))`
`Set-PSUCache -Key "Cache_RVTools_vNetwork" -Value ($RvTools_Import_vNetwork) -Persist -AbsoluteExpirationFromNow ([TimeSpan]::FromHours(25))`

![image](https://github.com/user-attachments/assets/6fbe6f38-2ee5-45e6-b4a7-7273089a9342)

![image](https://github.com/user-attachments/assets/fd718970-4f5b-40fd-bb6c-d2cb49c9a4d1)

**Comment by rwidmark** [comment]
Got the same issue, just want to write that so you know it's nothing in your setup.

**Comment by rwidmark** [comment]
#3722 I think that all of the issues are related.
In my case above I also did send to cavhe

**Comment by adamdriscoll** [comment]
If you haven't done so, you will either need to turn on the permissive security model (reverts to v4 behavior) or specify an app token. 

https://docs.powershelluniversal.com/config/module#authorization-security-model

**Comment by rwidmark** [comment]
> If you haven't done so, you will either need to turn on the permissive security model (reverts to v4 behavior) or specify an app token. 
> 
> 
> 
> https://docs.powershelluniversal.com/config/module#authorization-security-model

Even if you run the script from schedule?
I'm running it without a specific user so I'm guessing it run as the psuservice account

**Comment by emoeckel** [comment]
> If you haven't done so, you will either need to turn on the permissive security model (reverts to v4 behavior) or specify an app token.
> 
> https://docs.powershelluniversal.com/config/module#authorization-security-model

I followed this article and enabled permissive and trust certificate. The schedule trigger works now without an error. However, I am getting a "message size" error, when I try and hit the API that is serving up the cache.

My appsettings.json

`{
  "Kestrel": {
    "Endpoints": {
      "HTTP": {
        "Url": "http://*:443"
      }
    }
  },
  "Plugins": [
    "SQLite"
  ],
  "Data": {
    "RepositoryPath": "%ProgramData%\\UniversalAutomation\\Repository",
    "ConnectionString": "Data Source=%ProgramData%\\UniversalAutomation\\database.db"
  },
  "Mode": "Server",
  "PSUTelemetry": false,

   "Api": {
       "SecurityModel": "Permissive",
       "TrustCertificate": true
    }


}`



formatting always trips me up lol...
image of appsettings
![image](https://github.com/user-attachments/assets/80e6ce10-9cdf-478b-8f56-cf841dde4a57)




PowerShell from PC/Client

`$Test = Invoke-RestMethod http://igqprmgtpsu01:443/RVTools_vInfo -Method GET -SkipCertificateCheck`

Invoke-RestMethod: Status(StatusCode="ResourceExhausted", Detail="Received message exceeds the maximum configured message size.")
at , : line 2
at , : line 1

**Comment by emoeckel** [comment]
From the log for the above message size error

[9/11/2024 9:52:13 AM] [Information] [Api-1] Status: 400 Content-Type: application/json Length: 186 

[9/11/2024 9:52:13 AM] [Error] [Api-1] Status(StatusCode="ResourceExhausted", Detail="Received message exceeds the maximum configured message size.") 

[9/11/2024 9:52:13 AM] [Information] [Api-1] GET /RVTools_vInfo ::ffff:10.21.113.85

**Comment by rwidmark** [comment]
@adamdriscoll The thing above did solve it for powershell 7 but for PS5 I get the following error:
```
The term 'Set-PSUCache' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
The script failed due to call depth overflow.
```

Running PSU 5.0.7

**Comment by rwidmark** [comment]
@adamdriscoll I did try to import Universal Module both in environments.ps1 and directly in the script without any luck sadly.

**Comment by adamdriscoll** [comment]
This issue will be fixed in tonight's build.

**Comment by emoeckel** [comment]
> This issue will be fixed in tonight's build.

@adamdriscoll  sorry just to clarify, what issue is being fixed? This thread has gotten messy and not sure who you are replying to in this case. Thanks!

**Comment by adamdriscoll** [comment]
@emoeckel - I agree. I got lost too. The issue with the WinPS module path is resolved. 

This should have already been resolved in 5.0.6. 

Invoke-RestMethod: Status(StatusCode="ResourceExhausted", Detail="Received message exceeds the maximum configured message size.")

I cannot reproduce this myself. How large of an item is being cached? That said, it shouldn't matter because we completely removed the limit. I've tested with 100mb files and haven't had a problem
